client/state/data-layer/wpcom/sites/comments/test/index.js

```diff
@@ -76,58 +66,54 @@ describe( '#addComments', () => {
 			{ ID: 2, post: { ID: 1 } },
 		];
 
-		addComments( { dispatch }, { query }, { comments } );
-
-		expect( dispatch ).to.have.been.calledThrice;
+		const result = addComments( { query }, { comments } );
 
-		expect( dispatch ).to.have.been.calledWithMatch( {
+		expect( result[ 1 ] ).toEqual( {
+			siteId: query.siteId,
+			commentById: false,
+			type: COMMENTS_RECEIVE,
 			postId: 1,
 			comments: [ comments[ 0 ], comments[ 2 ] ],
 		} );
 
-		expect( dispatch ).to.have.been.calledWithMatch( {
+		expect( result[ 2 ] ).toEqual( {
+			siteId: query.siteId,
+			commentById: false,
+			type: COMMENTS_RECEIVE,
 			postId: 2,
 			comments: [ comments[ 1 ] ],
 		} );
 	} );
 } );
 
 describe( '#fetchCommentList', () => {
-	let dispatch;
-
-	beforeEach( () => ( dispatch = spy() ) );
-
 	test( 'should do nothing if no listType provided', () => {
-		fetchCommentsList( { dispatch }, { query } );
-
-		expect( dispatch ).to.have.not.been.called;
+		expect( fetchCommentsList( { query } ) ).not.toBeDefined();
 	} );
 
 	test( 'should do nothing if invalid listType provided', () => {
-		fetchCommentsList( { dispatch }, { query: { ...query, listType: 'Calypso' } } );
-
-		expect( dispatch ).to.have.not.been.called;
+		expect( fetchCommentsList( { query: { ...query, listType: 'Calypso' } } ) ).not.toBeDefined();
 	} );
 
 	test( 'should dispatch HTTP request for site comments', () => {
-		fetchCommentsList( { dispatch }, { query: { ...query, listType: 'site' } } );
-
-		expect( dispatch ).to.have.been.calledWithMatch( {
-			type: 'WPCOM_HTTP_REQUEST',
-			path: '/sites/1337/comments',
-		} );
+		expect( fetchCommentsList( { query: { ...query, listType: 'site' } } ) ).toEqual(
+			expect.objectContaining( {
+				type: 'WPCOM_HTTP_REQUEST',
+				path: '/sites/1337/comments',
+			} )
+		);
 	} );
 
 	test( 'should default to fetch pending comments', () => {
-		fetchCommentsList( { dispatch }, { query: { listType: 'site', siteId: 1337 } } );
-
-		expect( dispatch ).to.have.been.calledWithMatch( { query: { status: 'unapproved' } } );
+		expect( fetchCommentsList( { query: { listType: 'site', siteId: 1337 } } ) ).toEqual(
+			expect.objectContaining( { query: expect.objectContaining( { status: 'unapproved' } ) } )
```

Comment:
`.toMatchObject( { query: { status: 'unapproved' } } )` should do the same job.

Reply:
Unfortunately this doesn't work as we only match a subset of the object. `toMatchObject` expects any keys to be the same.

Reply:
Are you sure? The following test succeeds for me:
```js
expect( { a: { b: 1, c: 2 }, d: 3 } ).toMatchObject( { a: { b: 1 } } );
```
The values of `a` are different, but the expected subkeys are present.

The [toMatchObject](https://jestjs.io/docs/en/expect.html#tomatchobjectobject) docs also say that it should work. See the example with matching `houseForSale` against `desiredHouse` and how it matches although the `kitchen.area` key is not present in the `desiredHouse` object.

Reply:
You're right! I was thinking in the wrong direction 👍